src/db.c

```diff
@@ -860,6 +888,46 @@ int parseScanCursorOrReply(client *c, robj *o, unsigned long *cursor) {
     return C_OK;
 }
 
+char *obj_type_name[OBJ_TYPE_MAX] = {
+    "string", 
+    "list", 
+    "set", 
+    "zset", 
+    "hash", 
+    "", /* module type is special */
+    "stream"
+};
+
+long long getObjectTypeByName(char *name) {
+    for (long long i = 0; i < OBJ_TYPE_MAX; i++) {
+        if (!strcasecmp(name, obj_type_name[i])) {
+            return i;
```

Comment:
it will return 5 if `name` is '\0' or `\0xxxx`.
Perhaps `obj_type_name` can be changed to
```c
    "hash", 
    NULL, /* module type is special */
    "stream"
```

Reply:
but null pointer can not be used in strcasecmp funtion. I think your question is same with oranagra's https://github.com/redis/redis/pull/12209#discussion_r1233373737 , I prefer to compare strings ahead of time if they are empty in `getObjectTypeByName` 

```c
long long getObjectTypeByName(char *name) {
    if(strlen(name)==0) return LLONG_MAX; <------------------ new add

    for (long long i = 0; i < OBJ_TYPE_MAX; i++) {
        if (!strcasecmp(name, obj_type_name[i])) {
            return i;
        }
    }
```

Reply:
i think both of these cases are handled now since we use `strlen` (not `sdslen`), which means that even `\0xxxx` will exit this function early.
@sundb do you still see a problem?

p.s. maybe it's a good idea to use `NULL` as you suggested too, and add explicit code to avoid using it when NULL (before calling `strcasecmp`).

Reply:
ooh, I get you point. NULL is indeed better than a weird empty string. I'll do it tomorrow. thank you, I like the advice, guys.